src/Build/Collections/CopyOnReadEnumerable.cs

```diff
@@ -28,6 +29,8 @@ internal class CopyOnReadEnumerable<T> : IEnumerable<T>
         /// </summary>
         private readonly object _syncRoot;
```

Comment:
You can eliminate this field now and use the RW lock? It doesn't matter what the object is so long as it's readonly and private. That will claw back the size increase in this object form adding the RW lock field.

Reply:
Disregard, I didn't view the full diff. You still use that syncroot in another path.